simulation-scripts/perturb.sh

```diff
@@ -51,68 +38,61 @@ THIS_SCRIPT="${DIR}/$(basename "$0")"
 
 # required defaults
 declare -a ARGUMENTS
-EXPECTED_NUM_ARGUMENTS=1
 ARGUMENTS=()
 SCENARIO=''
 MASTER_HOST=""
 SLAVE_HOSTS=""
 BASTION_HOST=""
-IS_TEST_ONLY=0
-IS_FORCE=0
 
 main() {
 
-  [[ $# = 0 && "${EXPECTED_NUM_ARGUMENTS}" -gt 0 ]] && usage
+  [[ $# = 0 ]] && usage
 
-  parse_arguments "$@"
-  validate_arguments "$@"
+  while [ $# -gt 0 ]; do
```

Comment:
Disagree with undoing functional decomposition as it reduces code clarity and increases the effort to parse the main function.

Reply:
I think as the command line args have been dramatically reduced here having this inline with the main function is ok.  Stops have to stop between 2 functions to figure out what going on for someone new to the code.

Reply:
put parse_arguments function back, validate_argument seem a tad pointless as it would only be 3 lines